### ISSUE TITLE
Potential anti-cheat?

### ISSUE THREAD

**Issue by Fevix** [goatfungus/NMSSaveEditor#1074]
I've been unable to claim items from the Expedition or Quicksilver vendors for a while now, even when running without mods. It's possible I may have triggered some form of anti-cheat in No Man's Sky.

**Comment by Fevix** [comment]
First confirmation: After starting a new save and getting to the Anomaly legitimately without any cheats, I was able to successfully claim expedition rewards from the Quicksilver vendor. I will now start further testing.

**Comment by Fevix** [comment]
Tested: 

- Maximixed Units, Nanites, and Quicksilver (All to 4294967295)
- Gave myself ALL technologies and Known Products

Result: Unable to claim any further rewards

**Comment by Fevix** [comment]
Test: Maximize Units, Nanites, and Quicksilver
Result: Able to claim rewards

**Comment by Fevix** [comment]
I'm now going through and claiming the Known Products chunk by chunk

**Comment by Fevix** [comment]
I have confirmed that there is something in the "Known Products" section that, when learned, prevents you from claiming rewards from the Quicksilver Vendor.

**Comment by Fevix** [comment]
Looking more closely, in the Curiosities section, all of the rewards are there. Presumably this is used to track which rewards the player has claimed. I'd recommend blocking these rewards from being manually added.

**Comment by Fevix** [comment]
Removing all expedition rewards from my main save's Known Products did not, however, allow me to claim all rewards. Most, but not all. I was only able to claim one part of the Aquarius armor, and one part of the Chitin.

**Comment by tehface** [comment]
i can claim them all when i put steam to offline, so there ya go

**Comment by Fevix** [comment]
> i can claim them all when i put steam to offline, so there ya go

Please reread the entire first message. 

I'm not talking about how the game will reset your twitch rewards whenever you're connected to the internet 

I'm talking about how if you add the twitch rewards and the expedition rewards to your known products list, it prevents you from claiming them even if you've never claimed them before

**Comment by tehface** [comment]
because your " account data " is on their servers, so as i said to get around this, offline mode

**Comment by Fevix** [comment]
No, only the Twitch rewards are on their servers. If you remove the Expedition rewards from your Known Products you can claim them

**Comment by Devilin-Pixy** [comment]
Ever since the AccountData file has been introduced, users are expected to claim account wide awards on the actual Account tab. Failing to do so, can indeed causes issues as you are describing. At the same time, I do agree this is not made clear in any way using this tool. I'd love to see a clear warning for that matter. Then again, some understanding is also expected when making use of this tool. The sections provided to us on the regular save data, are useful, including the Discovery tab you are running into issues with. I don't believe removing the IDs from those sections to be a solution.

With that being said, the way the game verifies what rewards are available to you, is partly coming from the same section you are making changes to, as you have described. There is however more to it than just that single section, which is however not available to you within the regular use of the tool. There are additional sections in the save data as well, used for tracking claimed rewards, which you can edit through the Raw JSON.
Depending on active Context (Main/Season):
- PlayerStateData > RedeemedSeasonRewards
- PlayerStateData > RedeemedTwitchRewards
- PlayerStateData > RedeemedPlatformRewards
- PlayerStateData > KnownSpecials

With you adding all the possible tech and items as 'known', it did just that. This does not physically give you anything, like the Iron Vulture you used for an example. Yes, you now 'know' about the Iron Vulture now, but it has not given you the physical ship. Instead, it now causes the Shop where you are expected to get the ship from, to no longer give it to you, because 'knowing' about it, is part of the check done at the shop for claiming it. I get it though, I can't really blame you for wanting to have it, or to want it all for that matter. Nor can I blame you for not quite understanding how it is supposed to work, as this is not clearly described or warned for.

Now to fix it, you can either use the backup that was created when you first made those changes, or do it manually. Using the backup is the easiest way to do this. Doing this manually requires using the JSON editor, to remove the failing associated reward IDs (if present) from all of the above (relevant) sections. After having fixed it correctly, you can use the account tab instead, to then claim from the Shop on the Anomaly, making sure to go offline when it involves Twitch rewards.

Meantime we can hope the developer of this tool is willing to make things a bit clearer when it comes to claiming rewards.
PS: Some platform specific/special bonus items may still require to be added to the GCUSERETTINGSDATA.MXML to actually work (ie. exclusive Nintendo Switch ship)

**Comment by Fevix** [comment]
Okay so again someone misses the problem here

The actual in-game problem I was experiencing that led me to open this
issue was that all of my Expedition rewards were saying they were
available, but when I tried to claim them, the interaction circle filled up
and didn't claim. I know that you need to go to the account tab to make
things available to claim, I had already done that, I had already made
every reward available to claim, but at the same time I had learned every
single item in the "Known Products" list.

If you look through the known products list, You will find the Expedition
and twitch rewards. If the player has these items in their known products
list, then it doesn't matter whether or not they are eligible to claim the
item, they can't.

**Comment by Fevix** [comment]
> > Okay so again someone misses the problem here
> 
> I am not missing the problem. Have you actually checked the Raw JSON to see if 'KnownSpecials' now has those IDs you have been adding to your Known Products? Because that is where those IDs are most likely also appearing now? And that is why you will no longer be able to claim them, even though they show as available at the shop

I don't need to check the raw json to see them there. I can clearly see them myself in the known products list in the Discoveries tab.

**Comment by Devilin-Pixy** [comment]
> I do indeed believe that the best solution here is for the associated items to be removed from the items available to add to the known products list. This would completely prevent players from locking themselves out, because the only purpose adding these items to the known products list serves is to block claiming in the manner that I had encountered.

Although I would love to see users no longer trying to unlock rewards through the save, I disagree with removing the IDs from that list. I'd love to no longer have to help out those who, like you, started out the wrong way. I regularly run into this as an Atlas Emissary on the NMS Discord.

> I don't need to check the raw json to see them there. I can clearly see them myself in the known products list in the Discoveries tab.

I am just trying to make you aware of the fact there is more going on behind the scenes than you are able to see from the regular functionality provided. It often ends up getting messy with players having tried to fix it after having used both the save and account. That is where doing it through the Raw JSON becomes useful and at times even required. I am happy to know you did manage to get it sorted though.

**Comment by Fevix** [comment]
> I disagree with removing the IDs from that list. 
I would actually love to hear your reasoning behind this. As far as I can tell the only purpose leaving this in serves is as a trap for players like me that don't know better, for players that don't know exactly which products they shouldn't be adding. My argument for removal is actually exactly what you said me the after:

> I'd love to no longer have to help out those who, like you, started out the wrong way
By removing these items from the list, players will be free to just select all products (Ctrl+A) and add them all as a single click, while still being able to claim their expedition rewards.

**Comment by Devilin-Pixy** [comment]
> I would actually love to hear your reasoning behind this. As far as I can tell the only purpose leaving this in serves is as a trap for players like me that don't know better, for players that don't know exactly which products they shouldn't be adding.

For me it is just because it allows me to have quick access to them all, without exclusion. There have been times, even though uncommon, it has been useful to me and possibly others. Like you, after correcting in that same list, it does revert 'properly', also removing the 'KnownSpecials' entry if it has one associated. So even though it may have started the issue to begin with, having it available to remove there, might be useful at times. To me it is also useful as a reference list, for testing purposes, or to save myself time when adding something manually, even if it requires more to actually get something in my possession.

I think the main issue here, is not being aware of how it actually works with account wide rewards. Being pointed towards the Account tab for this, to actually let the game handle the various required steps, claiming the selected unlocks at the shop, will certainly help a lot. The listing of 'known' IDs is just a small overlapping part of functionality in-game. Might even consider having a separate listing for 'Known Specials' instead (which likely results in some duplicates).

**Comment by Fevix** [comment]
I wasn't advocating for the removal of it in the list entirely, only in the list of things you can ADD. I do agree that it should be available to remove, just not add in that list. 

Perhaps remove them from the Discovery tab entirely, and instead add a second checkbox to the lists in Account to mark something as Claimed?
